plugins.rst

```diff
@@ -26,66 +59,55 @@ The ``list`` command prints the currently installed plugins.
-also distribute the source code of their plugins. A plugin can be
-compiled from its source code with the ``compile`` command. A sample
-plugin ``test-plugin`` is included with the {Project} source code.
+Plugins are packaged and distributed as binaries encoded with the versatile
+Singularity Image Format (SIF). However, plugin authors may also distribute the
+source code of their plugins. A plugin can be compiled from its source code with
+the ``compile`` command. A number of example plugins are included in the
+``examples/plugins`` directory of the {Project} source.
 
 .. code::
 
-   $ {command} plugin compile examples/plugins/test-plugin/
+   $ {command} plugin compile examples/plugins/cli-plugin/
+   INFO:    Plugin built to: /home/dtrudg/Git/apptainer/examples/plugins/cli-plugin/cli-plugin.sif
 
-Upon successful compilation, a SIF file will appear in the directory of
-the plugin's source code.
+Upon successful compilation, a SIF file will appear in the directory of the
+plugin's source code.
 
 .. code::
 
-   $ ls examples/plugins/test-plugin/ | grep sif
-   test-plugin.sif
+   $ ls examples/plugins/cli-plugin/ | grep sif
+   cli-plugin.sif
 
 .. note::
 
-   Currently, **all** plugins must be compiled from the {Project}
-   source code tree.
-
-   Also, the plugins mechanism for the Go language that {Project} is
-   written in is quite restrictive - it requires extremely close version
-   matching of packages used in a plugin to the ones used in the program
-   the plugin is built for. Additionally {Project} is using build
-   time config to get the source tree location for ``{command} plugin
-   compile`` so that you don't need to export environment variables etc,
-   and there isn't mismatch between package path information that Go
-   uses. This means that at present you must:
-
-   -  Build plugins using the exact same version of the source code, in
-      the same location, as was used to build the {Project}
-      executable.
-
-   -  Use the exact same version of Go that was used to build the
-      executable when compiling a plugin for it.
+   Due to the structure of the {Project} project, and the strict
+   requirements of Go plugin compilation, **all** plugins must be compiled from
+   within the {Project} source code tree. 
+   
+   The ability to compile plugins outside of the {Project} tree, that
+   previously existed, has been removed due to incompatible changes in Go 1.18.
 
 Every plugin encapsulates various information such as the plugin's
 author, the plugin's version, etc. To view this information about a
 plugin, use the ``inspect`` command.
 
 .. code::
 
-   $ {command} plugin inspect examples/plugins/test-plugin/test-plugin.sif
-   Name: sylabs.io/test-plugin
-   Description: This is a short test plugin for {Project}
-   Author: Michael Bauer
-   Version: 0.0.1
+   $ {command} plugin inspect examples/plugins/cli-plugin/cli-plugin.sif
+   Name: github.com/{orgrepo}/cli-example-plugin
```

Comment:
Replace `github.com/{orgrepo}/cli-example-plugin` with `example.com/cli-plugin`.  The code actually currently does use the former, but it also affects all the references below which get awfully long, so we will plan to change the code via apptainer/apptainer#492.